File: code_samples/scala/algorithms_searching.scala
/**
 * Algorithms: Searching
 * AI/ML Training Sample
 */

class Searching {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object SearchingExample {
  def main(args: Array[String]): Unit = {
    val instance = new Searching()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/elixir/algorithms_searching.ex
# Algorithms: Searching
# AI/ML Training Sample

defmodule Searching do
  defstruct data: ""
  
  def new(), do: %Searching{}
  
  def process(%Searching{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Searching{data: data}), do: data
  
  def validate(%Searching{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Searching.new()
updated = Searching.process(instance, "example")
IO.puts("Data: " <> Searching.get_data(updated))
IO.puts("Valid: " <> to_string(Searching.validate(updated)))
